lib/vrl/stdlib/src/parse_ruby_hash.rs

```diff
@@ -0,0 +1,264 @@
+            char('{'),
+            cut(terminated(
+                map(
+                    separated_list0(preceded(sp, char(',')), parse_key_value),
+                    |tuple_vec| tuple_vec.into_iter().collect(),
+                ),
+                preceded(sp, char('}')),
+            )),
+        ),
+    )(input)
+}
+
+fn parse_value<
+    'a,
+    E: ParseError<&'a str> + ContextError<&'a str> + FromExternalError<&'a str, ParseIntError>,
+>(
+    input: &'a str,
+) -> IResult<&'a str, Value, E> {
+    preceded(
+        sp,
+        alt((
+            parse_nil,
+            map(parse_hash, Value::Object),
+            map(parse_array, Value::Array),
+            map(parse_bytes, Value::Bytes),
+            map(double, |value| Value::Float(NotNan::new(value).unwrap())),
+            map(parse_boolean, Value::Boolean),
+        )),
+    )(input)
+}
+
+fn parse(input: &str) -> Result<Value> {
+    let (rest, result) = parse_value(input).map_err(|err| match err {
+        nom::Err::Error(err) | nom::Err::Failure(err) => {
+            // Create a descriptive error message if possible.
+            nom::error::convert_error(input, err)
+        }
+        _ => format!("{}", err),
+    })?;
+
+    if rest.trim().is_empty() {
+        Ok(result)
+    } else {
+        Err("could not parse whole line successfully".into())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_parse_empty_object() {
+        let result = parse("{}").unwrap();
+        assert!(result.is_object());
+    }
+
+    #[test]
+    fn test_parse_simple_object() {
+        let result = parse(
+            r#"{ "hello" => "world", "number" => 42, "float" => 4.2, "array" => [1, 2.3], "nope" => nil }"#,
+        )
+        .unwrap();
+        assert!(result.is_object());
+        let result = result.as_object().unwrap();
+        assert!(result.get("hello").unwrap().is_bytes());
+        assert!(result.get("number").unwrap().is_float());
+        assert!(result.get("float").unwrap().is_float());
+        assert!(result.get("array").unwrap().is_array());
+        assert!(result.get("nope").unwrap().is_null());
+    }
+
+    #[test]
+    fn test_parse_weird_format() {
+        let result = parse(r#"{hello=>"world",'number'=>42}"#).unwrap();
+        assert!(result.is_object());
+        let result = result.as_object().unwrap();
+        assert!(result.get("hello").unwrap().is_bytes());
+        assert!(result.get("number").unwrap().is_float());
+    }
+
+    test_function![
+        parse_ruby_hash => ParseRubyHash;
+
+        complete {
+            args: func_args![value: value!("{ \"test\" => \"value\", \"testNum\" => 0.2, \"testObj\" => { \"testBool\" => true } }")],
+            want: Ok(value!({
+                test: "value",
+                testNum: 0.2,
+                testObj: {
+                    testBool: true,
+                }
+            })),
+            tdef: TypeDef::new().infallible().object::<(), Kind>(type_def()),
```

Comment:
While using `type_def()` for repeated checks makes sense, you'll want at least _one_ test that actually validates that whatever `type_def()` returns is what we expect. In this case, I believe it's incorrect, since it only reports values as being either strings or arrays.